core/tests/protractor_desktop/ExplorationFeedbackTab.js

```diff
@@ -0,0 +1,192 @@
+    users.createUser(
+      'user2@ExplorationFeedback.com',
+      'learnerExplorationFeedback');
+  });
+
+  it('adds feedback to an exploration', function() {
+    var feedback = 'A good exploration. Would love to see a few more questions';
+    var feedbackResponse = 'Thanks for the feedback';
+
+    // Creator creates and publishes an exploration.
+    users.login('user1@ExplorationFeedback.com');
+    workflow.createAndPublishExploration(
+      EXPLORATION_TITLE,
+      EXPLORATION_CATEGORY,
+      EXPLORATION_OBJECTIVE,
+      EXPLORATION_LANGUAGE);
+    creatorDashboardPage.get();
+    expect(
+      creatorDashboardPage.getNumberOfFeedbackMessages()
+    ).toEqual(0);
+    users.logout();
+
+    // Learner plays the exploration and submits a feedback.
+    users.login('user2@ExplorationFeedback.com');
+    libraryPage.get();
+    libraryPage.findExploration(EXPLORATION_TITLE);
+    libraryPage.playExploration(EXPLORATION_TITLE);
+    explorationPlayerPage.submitFeedback(feedback);
+    users.logout();
+
+    // Creator reads the feedback and responds.
+    users.login('user1@ExplorationFeedback.com');
+    creatorDashboardPage.get();
+    expect(
+      creatorDashboardPage.getNumberOfFeedbackMessages()
+    ).toEqual(1);
+    creatorDashboardPage.navigateToExplorationEditor();
+
+    explorationEditorPage.navigateToFeedbackTab();
+    explorationEditorFeedbackTab.expectToHaveFeedbackThread();
+    explorationEditorFeedbackTab.readFeedbackMessages()
+      .then(function(messages) {
+        expect(messages.length).toEqual(1);
+        expect(messages[0]).toEqual(feedback);
+      });
+    explorationEditorPage.navigateToFeedbackTab();
+    explorationEditorFeedbackTab.sendResponseToLatestFeedback(feedbackResponse);
+    users.logout();
+  });
+
+  afterEach(function() {
+    general.checkForConsoleErrors([]);
+  });
+});
+
+describe('Suggestions on Explorations', function() {
+  var EXPLORATION_TITLE = 'Exploration with Suggestion';
+  var EXPLORATION_CATEGORY = 'Algorithms';
+  var EXPLORATION_OBJECTIVE = 'To explore something new';
+  var EXPLORATION_LANGUAGE = 'English';
+  var creatorDashboardPage = null;
+  var libraryPage = null;
+  var explorationEditorPage = null;
+  var explorationEditorFeedbackTab = null;
+  var explorationPlayerPage = null;
+
+  beforeEach(function() {
+    explorationEditorPage = new ExplorationEditorPage.ExplorationEditorPage();
+    explorationEditorFeedbackTab = explorationEditorPage.getFeedbackTab();
+    explorationPlayerPage = new ExplorationPlayerPage.ExplorationPlayerPage();
+    creatorDashboardPage = new CreatorDashboardPage.CreatorDashboardPage();
+    libraryPage = new LibraryPage.LibraryPage();
+  });
+
+  beforeEach(function() {
+    users.createUser(
+      'user1@ExplorationSuggestions.com',
+      'authorExplorationSuggestions');
+    users.createUser(
+      'user2@ExplorationSuggestions.com',
+      'suggesterExplorationSuggestions');
+    users.createUser(
```

Comment:
I think we can save on creating a user here. We can use `user2` to perform `user3`'s actions. What do you think?

Reply:
As mentioned below, will leave this out from this PR.